### PR TITLE
6698-filter-markets-tag

### PR DESCRIPTION
**Pull request by JohnDanz**
fixed an issue with tags not filtering when selected on the side menu, fixed another visual issue where tags were displaying for markets not present in the filtered view.

[Clubhouse Story](https://app.clubhouse.io/augur/story/6698/unable-to-filter-market-categories-by-tag)

## Submitter checklist
- [x] Test covering functionality added/fixed in
- Check the linked story includes the following:
  - [x] Steps to reproduce
  - [ ] Screenshot (If applicable)

## Reviewer Checklist
- [ ] Test/lint pass 
- [ ] Post merge, story marked complete

### CODE REVIEW

File: src/modules/filter-sort/helpers/filter-by-tags.js [hunk 1]
```diff
@@ -6,9 +6,9 @@ export default function filterByTags(tags, items) {
   return items.reduce((p, item, i) => {
     if (tags.every(filterTag =>
       item.tags.some((tag, tagIndex) =>
-        tagIndex !== 0 && tag === filterTag))
+        tag === filterTag))
     ) {
-      return [...p, i]
+      return [...p, items[i].id]
```
Comment (stephensprinkle) [inline src/modules/filter-sort/helpers/filter-by-tags.js]: This helper function should be returning the index of the item, not the id of the item as other components (namely the rest of the filter-sort module) is architected with the expectation that this just returns an array of matched indexes.

File: src/modules/markets/components/markets-view.jsx [hunk 2]
```diff
@@ -66,6 +68,14 @@ export default class MarketsView extends Component {
   }
 
   componentWillUpdate(nextProps, nextState) {
+    const newTags = decodeURIComponent(parseQuery(nextProps.location.search)[TAGS_PARAM_NAME] || '')
```
Comment (stephensprinkle) [inline src/modules/markets/components/markets-view.jsx]: No need for the `decodeURIComponent` here.

Also, I’d do a conditional around just `location.search` as parsing that is a performance hit on each update.

File: src/modules/markets/components/markets-view.jsx [hunk 3]
```diff
@@ -66,6 +68,14 @@ export default class MarketsView extends Component {
...
+    if (newTags.length) {
+      const marketsFilteredByTags = filterByTags(newTags.split(QUERY_VALUE_DELIMITER), nextProps.markets)
```
Comment (stephensprinkle) [inline src/modules/markets/components/markets-view.jsx]: Per my comments on `filterByTags`, you’ll need to update handling of this value accordingly.

File: src/modules/markets/components/markets-view.jsx (hunk 2 above)
Comment (JohnDanz) [reply src/modules/markets/components/markets-view.jsx]: i was finding there was a need as tags such as `US Politics` where coming in URI encoded like so `US%20Politics`

**Comment by stephensprinkle** [comment]
Per discussion, merging down as is as the utilization of the filter-sort module will require a broader refactor.

**Pull request merged**